app/mailers/activity_mailer.rb

```diff
@@ -5,11 +5,12 @@ class ActivityMailer < ApplicationMailer
   # Emails a recipient, informing him of an activity.
   #
   # @param [User] recipient The recipient of the email.
-  # @param object The object to be made available to the view, accessible using +@object+.
+  # @param [Course::Notification|UserNotification] notification The notification to be made
+  #   available to the view, accessible using +@notification+.
   # @param [String] view_path The path to the view which should be rendered.
-  def email(recipient, object, view_path)
+  def email(recipient, notification, view_path)
     @recipient = recipient
-    @object = object
+    @notification = notification
```

Comment:
`@object = @notification.activity.object`, view probably don't need the whole notification.